Subject: [PATCH] Using requested video scaling mode in NuPlayer

In comparison to AwesomePlayer, NuPlayer has lack of setting
the video scaling mode flag on event kWhatVideoSizeChanged.
That causes the issues with wrong video screen ratio, because
ACodec has hardcoded flag which was the last one set up for
particular native window object, even that flow goes back to NuPlayer.

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index 77b9799..6ef2b3a 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -981,6 +981,7 @@ void NuPlayer::onMessageReceived(const sp<AMessage> &msg) {
                 sp<AMessage> inputFormat =
                         mSource->getFormat(false /* audio */);
 
+                setVideoScalingMode(mVideoScalingMode);
                 updateVideoSize(inputFormat, format);
             } else if (what == DecoderBase::kWhatShutdownCompleted) {
                 ALOGV("%s shutdown completed", audio ? "audio" : "video");
